extensions/funqy/funqy-server-common/deployment/src/main/java/io/quarkus/funqy/deployment/FunctionScannerBuildStep.java

```diff
@@ -172,6 +172,12 @@ private static void generateDefaultConstructors(BuildProducer<BytecodeTransforme
             Set<ClassInfo> withoutDefaultCtor) {
 
         for (ClassInfo classInfo : withoutDefaultCtor) {
+            // don't generate constructor for normal scoped beans as the Quarkus Arc does that for us
```

Comment:
So in theory, custom scopes can be normal as well. On the other hand, this solution probably covers 99% of use cases. Also the `CustomScopeAnnotationsBuildItem` can only be used to test the scope annotations and not whether they represent a normal scope.

Reply:
Didn't know that, thank you.